components/heap/heap_trace_standalone.c

```diff
@@ -46,8 +56,13 @@ typedef struct {
 
 
 // Forward Defines
-static void remove_record(records_t *r, int index);
 static void heap_trace_dump_base(bool internal_ram, bool psram);
+static void linked_list_setup(records_t* rs);
```

Comment:
If possible, please try to avoid adding custom linked list implementation and use the one provided by sys/queue.h, instead. ([Docs](https://www.freebsd.org/cgi/man.cgi?query=queue&sektion=3))

Reply:
yes I did find that implementation afterwards

I'll give it a shot